pkg/k2/orchestration/orchestrator.go

```diff
@@ -87,7 +87,15 @@ func (o *Orchestrator) resolveInitialState(ir *model.ApplicationEnvironment) (ma
 
 	// Check for default region mismatch
 	if state.DefaultRegion != ir.DefaultRegion {
-		return nil, fmt.Errorf("default region mismatch: %s != %s", state.DefaultRegion, ir.DefaultRegion)
+		deployed := make(map[string]model.ConstructStatus)
+		for k, v := range state.Constructs {
+			if model.IsDeletable(v.Status) {
+				deployed[k] = v.Status
+			}
+		}
+		if len(deployed) > 0 {
```

Comment:
we may need to couple this with deleting the old pulumi state

Reply:
I'm not too familiar with how state works, this should just be mostly the same as it was before. Can you explain a bit more what you mean and why it should delete old pulumi state?

Reply:
Sure, and it may not be pulumi state necessarily (it may be klotho2 state referencing non-existent resources since we never clear that file). We've seen cases where changing region and redeploying causes deploy time errors about vpcs etc. I was thinking we could start with modifying state to clear this out and see if that's enough.

Reply:
> changing region and redeploying causes deploy time errors

This should only happen if the klotho / pulumi state is out of sync with AWS, right? Since this checks whether any resources are destroyable before allowing the region swap. Any resource that was deployed and not properly cleaned up would still be in a destroyable state I'd think. If not, that sounds like there's a bug somewhere.

If that's that case, for the purposes of this PR, I'm leaning to it being out of scope and we should cut a new card for it.